### PR TITLE
Change SAC str representation

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
Add `report_id`
To represent submissions
Not just UEI.

-----

Changes how submissions look as strings (for example in the Django shell or in Django admin views) from e.g. `#4 - UEI(DJ4KNQ383PH7)` to  e.g. `#4--2023JUL0001000004--DJ4KNQ383PH7`.

### CODE REVIEW

**Review by danswick — APPROVED** [review]
Looks good!

File: backend/audit/test_models.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ class SingleAuditChecklistTests(TestCase):
     def test_str_is_id_and_uei(self):
         """String representation of SAC instance is #{ID} - UEI({UEI})"""
```
Comment (danswick) [inline backend/audit/test_models.py:24]:
```suggestion
    def test_str_is_id_report_id_and_uei(self):
        """String representation of SAC instance is #{ID}--{report_id}--{UEI}"""
```

Should we update these as well?

**Review by danswick — COMMENTED** [review]
Just a small suggestion

File: backend/audit/test_models.py (hunk 1 above)
Comment (tadhg-ohiggins) [reply backend/audit/test_models.py:24]: Whoops, yes, I should have added that too.

Comment (tadhg-ohiggins) [reply backend/audit/test_models.py:24]: I'll add that in another PR.

**Pull request merged**